### DISCUSSION TITLE
Docker support

### DISCUSSION THREAD

**Discussion by reality-exe** [glanceapp/glance#3] [General]
Docker support to be able to view the status of running docker containers would be really nice

**Comment by svilenmarkov** [comment]
Thanks for the suggestion.

I made the site monitor widget with the idea that you can easily check the status of your containers (at least the ones which have some kind of an HTTP server, which is most) that reside on the same server which Glance is running on as well as ones that are on other servers. I'm assuming you may have some containers that don't expose an HTTP server and would like to be able to keep an eye on them? I'll play around with Docker's API as well as some potential designs and see if I can come up with something.

**Reply by reality-exe** [reply]
Yeah I have a few things that don't have HTTP servers like Resonite headless' (Dedicated server for Resonite)

**Reply by varunpan** [reply]
check out https://github.com/Tecnativa/docker-socket-proxy

You can use this to check on docker container status. it would be ideal for monitor widget.

**Comment by shareef945** [comment]
i second this, so i'm currently using homepage for my dashboard and all i have to do when deploying a stack is add the relevant labels. 
<img width="1512" alt="Screenshot 2024-11-03 at 7 59 46 AM" src="https://github.com/user-attachments/assets/f47bafa5-d901-417d-89ef-7cb57c724641">

All i have to do when deploying is add the relevant labels like this and they automatically show on my homepage. 

`    labels:
      - homepage.group=Media
      - homepage.name=Transmission
      - homepage.icon=https://raw.githubusercontent.com/transmission/transmission/main/web/assets/img/logo.png
      - homepage.href=http://192.168.100.2:9091
      - homepage.description=Torrent download client`

**Comment by svilenmarkov** [comment]
@shareef945 There is a very nice 3rd party Glance extension that allows you to do this: 

https://github.com/DVDAndroid/glance-docker-container-ext

**Reply by shareef945** [reply]
sweet! Thanks a lot

**Reply by shareef945** [reply]
have you been able to get this running? i tried and the api response from the extension just says "No containers running"
